packages/ui/src/proposals/helpers/getDetailsRenderStructure.ts

```diff
@@ -171,14 +173,18 @@ const mapProposalDetail = (key: ProposalDetailsKeys, proposalDetails: ProposalWi
 }
 
 const getDetailsOrder = (proposalDetails: ProposalDetails): ProposalDetailsKeys[] => {
-  const definedOrders: { [key: string]: ProposalDetailsKeys[] } = {
+  const definedOrders: Partial<Record<ProposalType, ProposalDetailsKeys[]>> = {
     createWorkingGroupLeadOpening: ['group', 'stakeAmount', 'unstakingPeriod', 'rewardPerBlock', 'openingDescription'],
     decreaseWorkingGroupLeadStake: ['groupName', 'member', 'amount'],
     slashWorkingGroupLead: ['groupName', 'member', 'amount'],
+    updateWorkingGroupBudget: ['group', 'amount'],
   }
 
-  if (proposalDetails.type && definedOrders[proposalDetails.type]) {
-    return definedOrders[proposalDetails.type]
+  if (proposalDetails.type) {
+    const order = definedOrders[proposalDetails.type]
+    if (isDefined(order)) {
+      return order
+    }
```

Comment:
I don't see any change in logic here. I think that the result will be the same.

Reply:
This may be just a VSCode TS engine thing, but for some reason the typeguard used on an object member didn't work well and still saw `definedOrders[proposalDetails.type]` as array or undefined. Maybe because a string literal is used to index the object.